### PR TITLE
Apply redesign /partners/find-a-partner and add new /silicon partner page

### PR DESCRIPTION
**Pull request by mtruj013**
## Done

- Applied redesign as per [mock-up](https://www.figma.com/file/KQSCmTqOjY4gfKvJPyvzj2/Canonical.com-latest-only?type=design&node-id=9305-22310&t=9u07xXTxkSaNqaby-0)
- Added new Silicon page to partners bubble as per [design](https://www.figma.com/file/KQSCmTqOjY4gfKvJPyvzj2/Canonical.com-latest-only?type=design&node-id=9614%3A20510&t=cuBOBVZg6HBHGplX-1) 

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: https://canonical-com-894.demos.haus/partners/find-a-partner
- Check page against desgin
- Check filters and see that filter count is updated in the following ways:
   - If no filters are applied the text should read "All X partners"
   - If only one partner exists the text should read "X partner" (check this by typing "qualc" in the search bar, only QualComm should appear)
   - If there are no partners the text should read "0 partners" and there should be a notification prompting the user to search again
- Check filter state is maintained on refresh
- Check that on small screens the "Toggle filters" and "Clear filters" buttons show up and work as expected
  - Toggle filters should show/hide available filters
  - Clear filters should clear all selected filters
- Check that "Silicon" is a programme filter option (should yield 14 partners)
- Go to https://canonical-com-894.demos.haus/partners/silicon
  - See that page matches design  

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-3746, https://warthogs.atlassian.net/browse/WD-4330

### CODE REVIEW

**Comment by mtruj013** [comment]
@lyubomir-popov @juanruitina  this is ready for y'all to take a look for the most part, but there are some things I need a bit more guidance on:
- I saw on the mockup that there was [mention](https://www.figma.com/file/KQSCmTqOjY4gfKvJPyvzj2?node-id=9305:22310#457852277) of including the filter count, but I can't find where that is in the design. I'm not sure if I'm looking at the right version or not, but either way please let me know what you'd like me to do there. For now I've done this:  
image
- What is this page supposed to look like on medium and small screens? The mock-up does not cover this

Thanks!

**Comment by juanruitina** [comment]
Hi MariaPaula, such nice work. On your questions:

* Please just have "All [total] partners" by default, then change to "[Number] partners" (or "1 partner" when only one) when a search query or filter has been applied. Let's drop the blue chip.
* It generally looks good on mobile, but we need to do something about the filters. Same as done on the [Certified Hardware pages](https://ubuntu.com/certified?q=), I think we should hide them [in a drawer](https://vanillaframework.io/docs/patterns/navigation#side-navigation). The button should read "Toggle filter" same as there; next to it, we should also show a "Clear filters" button when some filter is applied, or users won't be able to tell users they are pre-filtered (it is common that we link to a filtered result page from partner pages).

I've put together a quick mockup [here on Figma](https://www.figma.com/file/KQSCmTqOjY4gfKvJPyvzj2/Canonical.com-latest-only?type=design&node-id=10191%3A25740&t=rBCFxTUmI62wCN0u-1) so it's clearer what I mean. @lyubomir-popov, can you have a look to make sure it makes sense to you too?

Some other minor stuff to tweak:

- [ ] Please add a slight margin above the logos so they don't get too close to the separator. In paragraphs that seems to be 0.4rem, and it's 8px on Figma.
- [ ] Can you please add a hidden label for the search field for screen readers? [From Vanilla docs](https://vanillaframework.io/docs/patterns/search-box): `<label class="u-off-screen" for="search">Search</label>` should do the trick.
- [ ] Change the [title of the partner links](https://github.com/canonical/canonical.com/blob/8bc7e0f7380cc99750d244c8f1b0a20c4505fca0/templates/partners/find-a-partner.html#LL187C31-L187C69) that head to their website to just `{{ partner['name'] }} website`
- [ ] Let's improve a bit the error message when there are no results. I checked with @sophie-32, let's change it to her suggestion:

> None of our partners match that search.
> Try another search or [check our full list of partners](https://canonical.com/partners/find-a-partner).

Thank you!

**Review by ClementChaumel — COMMENTED** [review]
There is also an issue with the query parameter. I don't think it's getting picked up on page load.
if you click on filters they get added to the query params like so https://canonical-com-894.demos.haus/partners/find-a-partner?filters=ihv-oem
but if you refresh you'll see that you'll have the filter unselected. This is an issue because we link to that page with filters from other pages

**Comment by lyubomir-popov** [comment]
@mtruj013 I've fixed all I needed code wise, but all logos need to be resized. This is blocked until we get them resized.

Some notes for future reference:

`<hr class="is-rule u-no-margin--bottom">`
should be simply
`<hr class="p-rule">`

please do not wrap checkboxes in paragraphs, see the vanilla example.

with vanilla 4.0, you can do `.p-muted-heading` instead of             
`p-text--x-small-capitalised u-align-text--x-small-to-default u-sv2`


this is pointless - one class adds padding-bottom, the other takes it away:
`              <div class="p-block u-no-padding--bottom">
`

**Comment by mtruj013** [comment]
@ClementChaumel @juanruitina @lyubomir-popov this is ready for you all to take a look at again, I've updated the QA instructions if you could please give that a read. I think I've covered everything, but there's a lot here so let me know if I missed something. 

In terms of the design I'm not sure what the correct one is anymore as the mock-up has been changed. Since Lyubo pushed his own updates I decided to take those as the source of truth and kept that mostly unchanged style-wise. The only difference there I think is the size of the logos. In the ticket the instructions were to set them to `max-width: 3rem; max-height: 1.5rem;` but it made them look absolutely minuscule so I thought this might be better. Let me know if you disagree and I'll change it.

**Comment by lyubomir-popov** [comment]
I pushed a couple of small fixes to make the logos less prominent, as the the random cropping otherwise interferes with the layout.

Looks great otherwise, thanks a lot @mtruj013.

**Comment by juanruitina** [comment]
- [ ] Filters column is to narrow and some labels wrap. Can we change it to `col-3`?
- [ ] Please leave the `alt` value for the logos empty.

On mobile:

- [ ] "Clear filters" should only be visible if a filter has been applied; I thinks it's always visible now.
- [ ] "Clear filters" clears the filter checkboxes, but doesn't reset the results as it should.
- [ ] I see the filters open right below, instead of in a drawer that slides from the side. Can you please add the full [side navigation](https://vanillaframework.io/docs/patterns/navigation#side-navigation) functionality, drawer included?
- [ ] I need to click twice on "Toggle filters" for them to show, can we fix this?

Also, most logos are barely readable now. I know this is tricky, but let's keep it in mind; they _should_ be readable.

**Comment by mtruj013** [comment]
> Filters column is to narrow and some labels wrap. Can we change it to col-3?

Not sure if I'm understanding what you mean here, it already is col-3:
Screenshot from 2023-06-26 14-42-09

> I see the filters open right below, instead of in a drawer that slides from the side. Can you please add the full [side navigation](https://vanillaframework.io/docs/patterns/navigation#side-navigation) functionality, drawer included?

Sorry, I completely misunderstood the ask here. Will update

> Also, most logos are barely readable now. I know this is tricky, but let's keep it in mind; they should be readable. Maybe increase min-height to 6rem instead of 4rem?

@lyubomir-popov would you be happy with this?

**Comment by juanruitina** [comment]
It's happening on a deeper level:

Screenshot 2023-06-26 at 16 19 05

**Comment by ClementChaumel** [comment]
Found a bug: if I type something that doesn't yield any result, it doesn't show the notification until I reload the page 
image

**Comment by mtruj013** [comment]
@ClementChaumel @juanruitina thanks all, I believe I've addressed everything if you could take another look?
___

Re: the nested col-2:

This was a change @lyubomir-popov [implemented](https://github.com/canonical/canonical.com/pull/894/commits/714c7a12fe4113d25259403121bc443ab1f4b0cc#diff-893348cc82d7ac5ef5a34aee699378e76ebee6f5b380634f336563e61e813945R44) I imagine because the medium screen view looks way better with it:

Screenshot from 2023-06-27 17-08-43

than without:
Screenshot from 2023-06-27 17-04-28

I won't update this until there's a clear decision agreed upon by both UX and Visual, please align on this and let me know what the outcome is.

**Comment by juanruitina** [comment]
I would actually hide the filters on medium viewpoints too, the drawer should work well, is that doable? Even if we don't, changing the filter blocks to `col-medium-2 col-3` should still make it work for all breakpoints.

I see in the mockup we don't have borders on top of the filter titles, and I don't think they work well in sidebars as they clash with the ones above each partner on the right. @lyubomir-popov would you agree with this?

Screenshot 2023-06-27 at 20 31 32

**Comment by lyubomir-popov** [comment]
> I see in the mockup we don't have borders on top of the filter titles, and I don't think they work well in sidebars as they clash with the ones above each partner on the right. @lyubomir-popov would you agree with this?

fine by me, will update

**Comment by lyubomir-popov** [comment]
@mtruj013 we've agreed with @juanruitina, see  above. let me know if you still need anything from our side here.

**Comment by mtruj013** [comment]
Thanks @lyubomir-popov and @juanruitina, updated. Juan please let me know if I have your +1 now.

**Comment by lyubomir-popov** [comment]
@mtruj013 the combination of min and max distorts some logos, can we please drop the height limit and just keep the max-width. We'll conduct a sprint to resize all those logos on Monday, so we should be ok.

image

**Comment by mtruj013** [comment]
@lyubomir-popov done

**Comment by lyubomir-popov** [comment]
Thank you! All good on my side.

**Comment by juanruitina** [comment]
In the demo I see filters still have classes `col-medium-2 col-2` and borders/separators at the top. I've also noticed logos overflow on mobile. Can you check?

canonical-com-894 demos haus_partners_find-a-partner(Figma mobile)

File: static/sass/styles.scss [hunk 1]
```diff
@@ -677,6 +677,103 @@ $p-small-lh-diff: map-get($line-heights, default-text) -
...
+    @media (min-width: $breakpoint-small) {
+      grid-column: auto / span 6;
+    }
+  }
+}
+
+.u-pad-logo {
+  padding-top: 0.2rem;
+}
+
```
Comment (ClementChaumel) [inline static/sass/styles.scss:776]: What is all this? It doesn't seem to get used here

File: templates/partners/find-a-partner.html [hunk 2]
```diff
@@ -2,254 +2,158 @@
...
+          </nav>
         </div>
       </div>
     </div>
+    <!-- End mobile side nav -->
   </section>
-  <section class="p-strip--light is-shallow js-search-form  u-hide" style="margin-top: -1px;">
+  <section class="p-section">
+    <hr class="p-rule is-fixed-width">
     <div class="row">
```
Comment (ClementChaumel) [inline templates/partners/find-a-partner.html:90]:
```suggestion
    <div class="row">
        <hr class="p-rule">
```

File: templates/partners/find-a-partner.html [hunk 3]
```diff
@@ -2,254 +2,158 @@
...
-            {% if partner['logo'] %}
-              <div style="height: 5rem; display: flex; align-items: flex-end;">
-                <img src="{{ partner['logo'] }}" alt="image for {{ partner['name'] }}" style="max-height: 5rem; max-width: 10rem;" loading="lazy">
+        <div class="col-9">
+          <div id="results" class="p-find-a-partner">
+            <div class="p-section">
+              <hr class="p-rule u-hide--large">
+              <h2 class="p-muted-heading" id="partners-count"></h2>
+            </div>
+            <div class="u-fixed-width">
```
Comment (ClementChaumel) [inline templates/partners/find-a-partner.html:116]:
```suggestion
            <div class="row">
```

File: templates/partners/silicon.html
Comment (ClementChaumel) [inline templates/partners/silicon.html:1]: Is this page supposed to be part of the find a partner PR?

**Review by ClementChaumel — APPROVED** [review]
LGTM but I think you might have some leftover code from other commits?

File: templates/partners/silicon.html [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+        <li class="p-list__item is-ticked">Security maintenance for 3rd party open source</li>
+        <li class="p-list__item is-ticked">Reference architecture and joint roadmap definition</li>
+        <li class="p-list__item is-ticked">Joint marketing and co-selling</li>
+      </ul>
+    </div>
+  </div>
+</section>
+
+<section class="p-section">
+  <div class="u-fixed-width">
```
Comment (ClementChaumel) [inline templates/partners/silicon.html:218]:
```suggestion
  <div class="row">
```

File: templates/partners/silicon.html [hunk 5]
```diff
@@ -0,0 +1,531 @@
...
+          <li class="p-list__item is-ticked"><a href="https://ubuntu.com/download/amd-xilinx">Install Ubuntu on AMD-Xilinx Zynq UltraScale+MPSoC-based evaluation boards and Kria SOMs</a></li>
+          <li class="p-list__item is-ticked"><a href="https://ubuntu.com/download/risc-v">Download Ubuntu for RISC-V Platforms</a></li>
+        </ul>
+      </div>
+    </div>
+  </div>
+</section>
+
+<section class="p-section">
+  <div class="u-fixed-width">
```
Comment (ClementChaumel) [inline templates/partners/silicon.html:286]:
```suggestion
  <div class="row">
```

File: templates/partners/find-a-partner.html (hunk 2 above)
Comment (lyubomir-popov) [reply templates/partners/find-a-partner.html:90]: is this for the hrs? It should preferably be u-fixed-width>hr, so it doesn't create a grid context

Comment (ClementChaumel) [reply templates/partners/find-a-partner.html:90]: The row is already there, why not use it?

**Comment by mtruj013** [comment]
@juanruitina I see what you mean now. Setting a max-height for logos is not a viable fix then, at least when it comes to the mobile view, as that's what's causing them to break out of the grid. Without it though, we're back to the issue of the logos being tiny and unreadable. Would you be completely against hiding them on small screens? (cc: @lyubomir-popov )

**Comment by mtruj013** [comment]
I already clarified [this](https://github.com/canonical/canonical.com/pull/894#discussion_r1246856161) with Clem via MM, but just wanted to leave an explanation here as well in case I've confused anyone else (sorry!). This PR covers updates to the `/find-a-partner` page as well as the `/silicon` partner page build. This is because part of the mvp for the find-a-partner page was to add a new programme filter (silicon) which the new partner page links to. I tried to leave that context in the pr description, but in retrospect I might've not done a great job of it 😅. I guess it would've also helped if I included the silicon page in the PR name, I'll update that now

**Comment by juanruitina** [comment]
I wonder if we can tweak it so max-width is 100% on smaller viewports and we keep whatever we have for bigger ones? I tried on the browser and it looks good enough.

**Comment by mtruj013** [comment]
UX +1 given on MM (Juan's traveling!), merging

**Pull request merged**